ext/vulnsrc/alpine/alpine.go

```diff
@@ -134,28 +139,65 @@ func detectNamespaces(path string) ([]string, error) {
 	return namespaces, nil
 }
 
-type parserFunc func(io.Reader) ([]database.Vulnerability, error)
+type lsFilter int
+
+const (
+	filesOnly lsFilter = iota
+	directoriesOnly
+)
+
+func ls(path string, filter lsFilter) ([]string, error) {
+	dir, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer dir.Close()
+
+	finfos, err := dir.Readdir(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var files []string
+	for _, info := range finfos {
+		if filter == directoriesOnly && !info.IsDir() {
+			continue
+		}
+
+		if filter == filesOnly && info.IsDir() {
+			continue
+		}
+
+		if strings.HasPrefix(info.Name(), ".") {
+			continue
+		}
+
+		files = append(files, info.Name())
+	}
 
-var parsers = map[string]parserFunc{
-	"v3.3": parse33YAML,
-	"v3.4": parse34YAML,
+	return files, nil
 }
 
 func parseVulnsFromNamespace(repositoryPath, namespace string) (vulns []database.Vulnerability, note string, err error) {
-	var file io.ReadCloser
-	file, err = os.Open(repositoryPath + "/" + namespace + "/main.yaml")
+	nsDir := filepath.Join(repositoryPath, namespace)
+	var dbFilenames []string
+	dbFilenames, err = ls(nsDir, filesOnly)
 	if err != nil {
 		return
 	}
-	defer file.Close()
 
-	parseFunc, exists := parsers[namespace]
-	if !exists {
-		note = fmt.Sprintf("The file %s is not mapped to any Alpine version number", namespace)
+	for _, filename := range dbFilenames {
+		var file io.ReadCloser
+		file, err = os.Open(filepath.Join(nsDir, filename))
+		if err != nil {
+			return
+		}
+		defer file.Close()
```

Comment:
please don't use `defer`, it will clog the fds, which won't be closed until the function ends. not like it matters too much considering the limited number of files in the Alpine's database but..